kolibri/core/assets/src/views/breadcrumbs/index.vue

```diff
@@ -0,0 +1,223 @@
+<template>
+
+  <nav class="breadcrumbs">
+
+    <div v-if="collapsedCrumbs.length" class="breadcrumbs-dropdown-wrapper">
+      <ui-icon-button :has-dropdown="true" icon="expand_more" size="small">
+        <div slot="dropdown" class="breadcrumbs-dropdown" :style="{ maxWidth: `${parentWidth}px` }">
+          <ol class="breadcrumbs-dropdown-items">
+            <li v-for="crumb in collapsedCrumbs" class="breadcrumbs-dropdown-item">
+              <router-link :to="crumb.link">{{ crumb.text }}</router-link>
+            </li>
+          </ol>
+        </div>
+      </ui-icon-button>
+    </div>
+
+    <ol class="breadcrumbs-visible-items">
+      <template v-for="(crumb, index) in visibleCrumbs">
+        <li
+          v-if="isLastCrumb(index)"
+          :ref="`visibleCrumb${index}`"
+          class="breadcrumbs-visible-item breadcrumb-visible-item-last"
+          :style="{ maxWidth: `${lastCrumbMaxWidth}px` }">
+          {{ crumb.text }}
+        </li>
+
+        <li v-else :ref="`visibleCrumb${index}`" class="breadcrumbs-visible-item breadcrumbs-visible-item-notlast">
+          <router-link :to="crumb.link">{{ crumb.text }}</router-link>
+        </li>
+      </template>
+    </ol>
+
+  </nav>
+
+</template>
+
+
+<script>
+
+  const ResponsiveElement = require('kolibri.coreVue.mixins.responsiveElement');
+  const ValidateLinkObject = require('kolibri.utils.validateLinkObject');
+  const filter = require('lodash/filter');
+  const startsWith = require('lodash/startsWith');
+  const throttle = require('lodash/throttle');
+
+  const DROPDOWN_WIDTH = 55;
+
+  module.exports = {
+    mixins: [ResponsiveElement],
+    $trNameSpace: 'breadcrumbs',
+    components: {
+      'ui-icon-button': require('keen-ui/src/UiIconButton'),
+    },
+    props: {
+      items: {
+        type: Array,
+        required: true,
+        validator(items) {
+          const crumbs = Array.from(items);
+          // Must not be empty
+          if (!crumbs.length) {
+            return false;
+          }
+          // All must have text
+          if (!crumbs.every(crumb => Boolean(crumb.text))) {
+            return false;
+          }
+          crumbs.pop();
+          // All, but the last, must have a valid router link
+          return crumbs.every(item => ValidateLinkObject(item.link));
+        },
+      },
+    },
+    data: () => ({
+      crumbs: [],
+    }),
+    computed: {
+      visibleCrumbs() {
+        return this.crumbs.filter(crumb => crumb.collapsed === false);
+      },
+      collapsedCrumbs() {
+        return this.crumbs.filter(crumb => crumb.collapsed === true).reverse();
+      },
+      parentWidth() {
+        return this.elSize.width;
+      },
+      lastCrumbMaxWidth() {
+        return this.parentWidth - DROPDOWN_WIDTH;
+      },
+    },
+    methods: {
+      isLastCrumb(index) {
+        return index === this.visibleCrumbs.length - 1;
+      },
+      crumbsPromise() {
+        return new Promise((resolve, reject) => {
```

Comment:
`this.$nextTick()` (with no callback argument) returns a promise. So you could probably do this

```
this.$nextTick().then(function filterCrumbs() { ... });
```